frontend/src/home/components/Homepage.jsx

```diff
@@ -25,26 +25,42 @@ export default class Homepage extends Component {
     }
 
     static propTypes = {
-        dispatch: PropTypes.func.isRequired,
         onChangeLocation: PropTypes.func.isRequired,
         showOnboarding: PropTypes.bool.isRequired,
-        user: PropTypes.object.isRequired
+        user: PropTypes.object.isRequired,
+        activity: PropTypes.array,
+        hasSampleDataset: PropTypes.bool,
+        fetchActivity: PropTypes.func.isRequired,
+        fetchRecentViews: PropTypes.func.isRequired,
+        fetchHasSampleDataset: PropTypes.func.isRequired
     };
 
+    async componentDidMount() {
+        try {
+            await this.props.fetchHasSampleDataset();
```

Comment:
I suppose, but it's not really much of a practical scenario.  It's not all that likely that someone goes into the admin section and deletes the sample dataset, then goes out of the admin section and tries to run the QB tutorial.

We can also protect against that by calling `MetabaseSettings.setAll({sample_data_available: false})` when the sample data is deleted.